fala/apps/adviser/tests/test_results_page_journeys_playwright.py

```diff
@@ -2,20 +2,16 @@
 from fala.playwright.setup import PlaywrightTestSetup
 
 
-class TestEndToEndJourneys(PlaywrightTestSetup):
-
-    def setUp(self) -> None:
-        self.page = self.browser.new_page()
-
-    def visit_valid_uk_results_page(self):
-        self.page.goto(f"{self.live_server_url}")
-        self.page.get_by_label("Postcode").fill("SE11")
-        self.page.get_by_role("button", name="Search").click()
-        expect(self.page.locator("h1")).to_have_text("Search results")
-
+class ResultPageEndToEndJourneys(PlaywrightTestSetup):
     def test_pagination(self):
-        self.visit_valid_uk_results_page()
-        expect(self.page.locator("span.govuk-pagination__link-title")).to_be_visible()
-        self.page.locator('span.govuk-pagination__link-title:has-text(" Next")').click()
-        expect(self.page.locator("h1")).to_have_text("Search results")
-        expect(self.page.locator('span.govuk-pagination__link-title:has-text(" Previous")')).to_be_visible()
+        page = self.visit_results_page("SE11")
+        expect(page.pagination_link_title).to_be_visible()
+        page.next_link.click()
+        expect(page.h1).to_have_text("Search results")
+        expect(page.previous_link).to_be_visible()
+
+    def test_change_search(self):
```

Comment:
we do already test this in the other region postcode tests but might be nice to check the persistence of the postcode here when they change search

Reply:
Frustratingly, testing this scenario (persisting postcode) is hard to get right. I think it's a limitation of Playwright. This is the addition we would want in the test:

`expect(search_page.postcode_input_field).to_have_text("SE11")`